test/spmd/test_xla_distributed_checkpoint.py

```diff
@@ -98,6 +106,11 @@ def test_sharded_to_unsharded(self):
     model = self.SimpleLinear().to(xm.xla_device())
     sharded_model = self._get_sharded_model()
     self._save_and_restore(sharded_model, model, save_planner=SPMDSavePlanner())
+    self._save_and_restore(
```

Comment:
We should re-initialize the models before running the test. If we can't parametrize the test, let's make this a separate test